### PR TITLE
Allow kwargs to pass to lldl

### PR DESCRIPTION
**Pull request by JeffFessler**
This should address #30.
I have not yet added and tests to confirm the additional functionality.
@learning-chip does this seem like the appropriate extension and can you suggest a test?

### CODE REVIEW

**Comment by mohdibntarek** [comment]
The implementation looks correct to me. Please add a test and let's this get this merged.

**Comment by JeffFessler** [comment]
I added a minimal test.
I also made it more clear to future developers which functions come from which packages.
There was also an unused `where T` that julia v1.8 complained about that I removed.
And I bumped the version to 0.6.

**Comment by JeffFessler** [comment]
@mohamed82008 this is ready for you to review

File: src/incompletecholesky.jl [hunk 1]
```diff
@@ -2,26 +2,36 @@ function assert_pd(d, α)
     @assert α == 0 && all(x -> x > 0, d) "The input matrix is not positive definite."
 end
 
+"""
+    CholeskyPreconditioner(A, memory=2; kwargs...)
+Construct
```
Comment (mohdibntarek) [inline src/incompletecholesky.jl:7]: We should use JuliaFormatter at some point to have consistent formatting. The lack of an empty line before `Construct` and short docstring lines strike me as odd but it's not a blocker. I am happy to follow any standard automated format.

**Review by mohdibntarek — APPROVED** [review]

**Pull request merged**